src/promise.js

```diff
@@ -14,19 +14,29 @@ var UNDEFINED_BINDING = {};
 var ASSERT = require("./assert");
 var util = require("./util");
 
-var getDomain;
+var getContext;
 if (util.isNode) {
-    getDomain = function() {
-        var ret = process.domain;
-        if (ret === undefined) ret = null;
-        return ret;
-    };
+    if (util.nodeSupportsAsyncResource) {
+        var AsyncResource = require("async_hooks").AsyncResource;
```

Comment:
Is this stripped somehow in the browser build? E.g. will this expression make it into `js/browser/bluebird.js`? If so, it might cause problems while bundling. Maybe we'd want to put this logic into a completely separate file that's "guarded" by a file-level browser override so it's excluded from bundling? E.g. something like this:

```
// src/getContext.browser.js, gets loaded directly in browser environments
module.exports = function getContext() { return {}; }

// src/getContext.js, gets loaded in not-definitely-browser environments
var getFallbackContext = require('./getContext.browser.js');
var getContext;
// [...]
```

Reply:
Yes, see the `exclude` in the browserify instructions, require will just return an empty object in that case (and even that won't be executed due to the isNode check before).

Reply:
Ah, gotcha. It compiles to `_dereq_('async_hooks')` which just never gets executed. 👍